pkg/operator/vspherecontroller/vspherecontroller.go

```diff
@@ -818,26 +819,64 @@ func getUserAndPassword(namespace string, secretName string, vcenter string, inf
 	// Get username and pass from secret created by CCO
 	username := string(secret.Data[usernameKey])
 	password := string(secret.Data[passwordKey])
+	return username, password, nil
+}
 
-	// The CSI driver expects a password with any quotation marks and backslashes escaped.
-	// xref: https://github.com/kubernetes-sigs/vsphere-csi-driver/issues/121
-	// The username in the format "domainName\userName" must be converted to "domainName\\userName"
-	// xref: https://docs.vmware.com/en/VMware-vSphere-Container-Storage-Plug-in/3.0/vmware-vsphere-csp-getting-started/GUID-BFF39F1D-F70A-4360-ABC9-85BDAFBE8864.html
-	return escapeBackslashInUsername(username), escapeQuotesAndBackslashes(password), nil
+// The CSI driver expects a password with any quotation marks and backslashes escaped.
+// xref: https://github.com/kubernetes-sigs/vsphere-csi-driver/issues/121
+// The username in the format "domainName\userName" must be converted to "domainName\\userName"
+// xref: https://docs.vmware.com/en/VMware-vSphere-Container-Storage-Plug-in/3.0/vmware-vsphere-csp-getting-started/GUID-BFF39F1D-F70A-4360-ABC9-85BDAFBE8864.html
+func escapeUserAndPassword(username, password string) (string, string, error) {
+	escapedUserName, err := escapeBackslashInUsername(username)
+	if err != nil {
+		return "", "", fmt.Errorf("error escaping username: %v", err)
+	}
+	escapedPassword, err := escapeQuotesAndBackslashes(password)
+	if err != nil {
+		return "", "", fmt.Errorf("error escaping password: %v", err)
+	}
+	return escapedUserName, escapedPassword, nil
 }
 
 // escapeQuotesAndBackslashes escapes double quotes and backslashes in the input string.
-func escapeQuotesAndBackslashes(input string) string {
-	return reEscape.ReplaceAllString(input, `\$0`)
+func escapeQuotesAndBackslashes(data string) (string, error) {
+	var js json.RawMessage
+	data = strings.ReplaceAll(data, "\n", " ")
+	jsonString := createJsonString(data)
+	byteValue := []byte(jsonString)
+	//escape special characters only if json unmarshall results in an error
+	if err := json.Unmarshal(byteValue, &js); err != nil {
+		var buf bytes.Buffer
+		encoder := json.NewEncoder(&buf)
+		// Disable HTML escaping to avoid conversion of &, >, < etc to unicode char sequences
+		encoder.SetEscapeHTML(false)
+		// encoder.Encode will escape special characters
+		err = encoder.Encode(data)
```

Comment:
I checked [JSON escaping](https://cs.opensource.google/go/go/+/master:src/encoding/json/encode.go;l=1020) and [gcfg escaping](https://github.com/openshift/vmware-vsphere-csi-driver-operator/blob/0d9a69f3e60f827e083934cc85141b7b1b4d1aa7/vendor/gopkg.in/gcfg.v1/read.go#L16), they're different when handling ASCII characters <32. I _think_ we can live with that - `\r`, `\b` or `\f` in a password?

Still, there should be at least a comment about that.